packages/@react-spectrum/dnd/docs/dnd.mdx

```diff
@@ -189,99 +207,241 @@ function DraggableList(props) {
+
+```tsx example
+function DroppableListLowLevelAPI() {
+  let list = useListData({
+    initialItems: [
+      {id: 1, name: 'Images', contains: 0, accept: ['image/png', 'image/jpeg']},
+      {id: 2, name: 'Videos', contains: 0, accept: ['video/mp4']},
+      {id: 3, name: 'Documents', contains: 0, accept: ['text/plain', 'application/pdf']}
+    ]
+  });
+
+  let {dndHooks} = useDnDHooks({
     onDrop: async (e) => {
-      let itemsToAdd = await Promise.all(
+      let items = await Promise.all(
         e.items
-          .filter(item => item.kind === 'text' && item.types.has('adobe-app'))
-          .map(async item => JSON.parse(await item.getText('adobe-app')))
+          .filter((item) => {
+            // Check if dropped item is accepted.
+            if (e.target.type === 'item' && e.target.dropPosition === 'on') {
+              let folder = list.getItem(e.target.key);
+              return folder.accept.includes(item.type);
+            }
+
+            return item.kind === 'directory';
+          })
+          .map(async (item) => {
+            // Collect child count from dropped directories.
+            let contains = 0;
+            let accept;
+            if (item.kind === 'directory') {
+              for await (let _ of item.getEntries()) {
+                contains++;
+                accept = [];
+              }
+            }
+
+            return {
+              id: Math.random(),
+              name: item.name,
+              contains,
+              accept
+            };
+          })
       );
-      list.append(...itemsToAdd);
+
+      // Update item count if dropping on an item, otherwise insert the new items in the list.
+      if (e.target.type === 'item') {
+        if (e.target.dropPosition === 'on') {
+          let item = list.getItem(e.target.key);
+          list.update(e.target.key, {
+            ...item,
+            contains: item.contains + items.length
+          });
+        } else if (e.target.dropPosition === 'before') {
+          list.insertBefore(e.target.key, ...items);
+        } else if (e.target.dropPosition === 'after') {
+          list.insertAfter(e.target.key, ...items);
+        }
+      } else {
+        // If dropping on the root, append the directory to the bottom of the list
+        list.append(...items);
+      }
+    },
+    getDropOperation: (target, types, allowedOperations) => {
+      // When dropping on an item, check whether the item accepts the drag types and cancel if not.
+      if (target.type === 'item' && target.dropPosition === 'on') {
+        let item = list.getItem(target.key);
+        return item.accept && item.accept.some((type) => types.has(type))
+          ? allowedOperations[0]
+          : 'cancel';
+      }
+
+      // If dropping a directory between items, support a copy operation.
+      return types.has(DIRECTORY_DRAG_TYPE) ? 'copy' : 'cancel';
     }
```

Comment:
I was unable to get dropping directories between items to work.

Reply:
derp, forgot to import the drag type, thanks for catching